introduction_to_applying_machine_learning/visual_object_detection/README.md

```diff
@@ -0,0 +1,145 @@
+2. [Solution Details](#solution)
+   1. [Background](#background)
+   2. [What is Visual Inspection?](#inspection)
+   3. [What are the Problems?](#problems)
+   4. [What Does this Solution Offer?](#offer)
+3. [Architecture Overview](#architecture)
+4. [Cleaning up](#cleaning-up)
+5. [Customization](#customization)
+
+
+## 1. Overview <a name="overview"></a>
+
+### 1.1. How Does the Input Data Look Like? <a name="input"></a>
+
+Input is an image of a defective / non-defective product. The training data should have relatively balanced classes, with annotations for ground truth defects (locations and defect types) per image. Here is an example used in the demo
+
+!["sample2"](https://sagemaker-solutions-prod-us-east-2.s3.us-east-2.amazonaws.com/sagemaker-defect-detection/docs/sample2.png)
+
+The NEU surface defect database (see [references](#references)) is a *balanced* dataset which contains
+
+> Six kinds of typical surface defects of the hot-rolled steel strip are collected, i.e., rolled-in scale (RS), patches (Pa), crazing (Cr), pitted surface (PS), inclusion (In) and scratches (Sc). The database includes 1,800 grayscale images: 300 samples each of six different kinds of typical surface defects
+
+Here is a sample images of the six classes
+
+!["data sample"](https://sagemaker-solutions-prod-us-east-2.s3.us-east-2.amazonaws.com/sagemaker-defect-detection/docs/data.png)
+
+### 1.2. How to Prepare Your Data to Feed into the Model? <a name="preparedata"></a>
+
+There are data preparation and preprocessing steps and should be followed in the notebooks. It's critical to prepare your image annotations beforehand.
+For finetuning pretrained Sagemaker models, you need to prepare either a single `annotation.json` for all data, or a `RecordIO` file for both all images and all annotations. Check the notebook for details.
+
+### 1.3. What are the Outputs? <a name="output"></a>
+
+* For each image, the trained model will produce bounding boxes of detected visual defects (if any), the predicted defect type and prediction confidence score (0~1).
+* If you have a labeled test dataset, you could obtain the mean Average Precision (mAP) score for each model and compare among all the models.
+   * For example, the mAP scores on a test set of the NEU dataset
+
+     |     |  Type1 | Type1+HPO | Type2 | Type2+HPO|
+     | --- |  --- | --- | --- | ---|
+     | mAP | 0.067 | 0.226 | 0.371 | 0.375|
+
+
+### 1.4. What is the Estimated Cost? <a name="cost"></a>
+
+* Running the notebook costs around $130~140 USD. This notebook provides advanced materials, including finetuning two types of pretrained Sagemaker models **till convergence**, with and without hyperparameter optimization (HPO), and result in four models for inference. You could choose to train either one model, or all four models according to your budget and requirements. The cost and runtime for training each model (assuming using p3.2xlarge EC2 instance) are:
```

Comment:
We typically don't try to quote costs in docs because they're region-specific and change w/o notice. It's very nice to provide estimates though. So make sure you say, for example, or if the cost is x in y region, then here's how you'd do the math, rather than say this "z", is the cost... because you just don't know.

Reply:
Very nice suggestion. I added the "if the cost is x in y region"-like description. Thanks